src/server/routes/helper.js

```diff
@@ -201,6 +203,152 @@ function exportApplicantInfo(users, archive, finalize) {
+    return writeToCSV(masterList);
+  }
+
+  let toBePicked = require(process.argv[2]);
+  let newMasterList = [];
+
+  toBePicked.forEach((x) => {
+    if (!indexMapper[x]) {
+      console.log("\x1b[31m", `X There was an email, ${x} in your file that there is no application for -- this should never happen` , "\x1b[31m");
+      return;
+    }
+    newMasterList.push(masterList[indexMapper[x]]);
+  })
+
+  return writeToCSV(newMasterList);
+};
+
+function bfs(vertices) {
+
+  /*vertices = {
+    'David': ['Panda'],
+    'Yacoub': ['Panda'],
+    'Nick' : ['Yacoub', 'David'],
+    'Panda' : [],
+  };*/
+ 
+  let visited = new Set(),
+    indexMapper = {},
+    masterList = [];
+
+    console.log('\x1b[32m', '✓ Building Teams' ,'\x1b[0m');
+
+  //for each node...
+  Object.keys(vertices).forEach((v, j) => {
+
+
+    let q = [v],
+      currTeam = new Set(),
+      flag = true,
+      info = 'Applications Not Found: ';
+
+    while (q.length !== 0) {
+      v = q.shift();
+
+      //don't revisit
+      if (visited.has(v)) {
+        continue;
+      }
+
+      visited.add(v);
+      currTeam.add(v);
+
+      //if an application for this email exists..
+      if (vertices[v]) {
+        //loop through children
+        vertices[v].forEach(c => {
+
+          if (visited.has(c)) {
+
+            //if we hit a child that exists in masterList
+            if (masterList[indexMapper[c]]) {
+
+              //update masterList and reflect the change in indexMapper
+              masterList[indexMapper[c]].team = masterList[indexMapper[c]].team.add(v);
+              indexMapper[v] = indexMapper[c];
+
+              //set flag to end BFS 
+              flag = false;
+            }
+            return;
+          }
+          else {
+
+            //unvisit node, enqueue it
+            q.push(c);
+          }
+        });
+      }
+      else {
+        //no application found
+        info += v + "     ";
+      }
+
+      //break out of the BFS
+      if (flag === false) {
+        break;
+      }
+    }
+
+    //these changes are already made in lines 74-76 for when flag = false
+    if (flag && currTeam.size > 0) {
+      currTeam.forEach(c => indexMapper[c] = masterList.length);
+      masterList.push({team: currTeam, info});
+    }
+
+    //write to CSV on end of this loop
+    if (j === Object.keys(vertices).length - 1) {
+      return cherryPick(masterList, indexMapper);
+    }
+  });
+};
+
+function buildVertices(users) {
+  console.log('\x1b[32m', '✓ Building Graph Vertices' ,'\x1b[0m');
+  let vertices = {};
+
+  users.forEach((user, i) => {
+    let currNodeChildren = [];
+    if (user.teammates && user.teammates.length > 0) {
+      user.teammates.forEach(x => {
+        if (x.length > 0) currNodeChildren.push(x.toLowerCase());
+      });
+    }
+    vertices[user.account.email.toLowerCase()] = currNodeChildren;
+
+    if (i === users.length - 1) {
+      return bfs(vertices);
+    }
+  });
+};
+
+
```

Comment:
Might be better to have all of these in a separate file. This is less of an API helper method and more of its own system.